### PR TITLE
fix(animations): only use the WA-polyfill alongside AnimationBuilder

### LINKED ISSUE #17496: Animation broken on ios7 safari 11 (ipad4-safari@browserstack)
## I'm submitting a ...
<pre><code>
[ ] Regression (behavior that used to work and stopped working in a new release)
[x] Bug report 
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
When i open my ng2 app on ipad, animations goes wrong (some styles don't applied on elements)

## Expected behavior
Everything works as expected (like on desktop)

## Minimal reproduction of the problem with instructions
Create animations and open it on ipad.
Animations:
```ts
@Component({
  ...
  animations: [
    trigger('color', [
      state('white', style({ backgroundColor: '#fff' })),
      state('red', style({ backgroundColor: '#e41e2b'})),
      transition('* => *', animate('250ms ease-in-out')),
    ]),
    trigger('visible', [
      state('hide', style({ opacity: 0, visibility: 'hidden' })),
      state('show', style({ opacity: 1, visibility: 'visible' })),
      transition('* => *', animate('250ms ease-in-out')),
    ]),
    trigger('size', [
      state('default', style({ transform: 'translateY(0)' })),
      state('large', style({ transform: 'translateY(11px)' })),
      transition('* => *', animate('250ms ease-in-out')),
    ]),
  ]
})
```

Template:
```html
<div class="action-button" [@size]="size" *ngIf="!isAuthPage">
  <div (click)="onBackButtonClick()" [@visible]="isShowBackButton" class="action-button__wrapper-back">
    <svg class="svg-icon action-button__back">
      <use xlink:href="./../../../assets/symbol/svg/sprite.css.svg#arrow-left"></use>
    </svg>
  </div>
  <ccff-burger [@visible]="isShowBurger" [isCrossed]="isMenuOpened" [color]="headerColor" (click)="onBurgerClick()"></ccff-burger>
</div>
```

Console results on ipad:
```
vendor.bundle.js:1229 ERROR Error__defineGetter__: __defineGetter__()__defineSetter__: __defineSetter__()__lookupGetter__: __lookupGetter__()__lookupSetter__: __lookupSetter__()constructor: Error()hasOwnProperty: hasOwnProperty()isPrototypeOf: isPrototypeOf()line: 11837line: 11837message: "Uncaught (in promise): Response with status: 401 Unauthorized for URL: /api/user/me"message: "Uncaught (in promise): Response with status: 401 Unauthorized for URL: /api/user/me"name: "Error"originalStack: "ZoneAwareError@http://127.0.0.1:3000/polyfills.bundle.js:11837:28↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11550:40↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11236:36↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11235:48↵runTask@http://127.0.0.1:3000/polyfills.bundle.js:11003:57↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44"originalStack: "ZoneAwareError@http://127.0.0.1:3000/polyfills.bundle.js:11837:28↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11550:40↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11236:36↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11235:48↵runTask@http://127.0.0.1:3000/polyfills.bundle.js:11003:57↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44"promise: ZoneAwarePromisepromise: ZoneAwarePromisepropertyIsEnumerable: propertyIsEnumerable()rejection: Responserejection: ResponsesourceURL: "http://127.0.0.1:3000/polyfills.bundle.js"sourceURL: "http://127.0.0.1:3000/polyfills.bundle.js"stack: "ZoneAwareError@http://127.0.0.1:3000/polyfills.bundle.js:11837:28↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11550:40↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11236:36↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11235:48↵runTask@http://127.0.0.1:3000/polyfills.bundle.js:11003:57↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44"stack: "ZoneAwareError@http://127.0.0.1:3000/polyfills.bundle.js:11837:28↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11550:40↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11236:36↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11235:48↵runTask@http://127.0.0.1:3000/polyfills.bundle.js:11003:57↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44"task: ZoneTasktask: ZoneTasktoLocaleString: toLocaleString()toString: toString()valueOf: valueOf()zone: Zonezone: ZonezoneAwareStack: "resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31 [angular]↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31 [angular]↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47 [angular]↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42 [<root>]↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44 [<root>]"zoneAwareStack: "resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31 [angular]↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31 [angular]↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47 [angular]↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42 [<root>]↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44 [<root>]"__proto__: Error__proto__: ErrordefaultErrorLogger @ vendor.bundle.js:1229handleError @ vendor.bundle.js:1289next @ vendor.bundle.js:4771(anonymous function) @ vendor.bundle.js:3844__tryOrUnsub @ vendor.bundle.js:26610next @ vendor.bundle.js:26559_next @ vendor.bundle.js:26512next @ vendor.bundle.js:26476next @ vendor.bundle.js:46023emit @ vendor.bundle.js:3830triggerError @ vendor.bundle.js:4202onHandleError @ vendor.bundle.js:4163handleError @ polyfills.bundle.js:11207runGuarded @ polyfills.bundle.js:10979_loop_1 @ polyfills.bundle.js:11442drainMicroTaskQueue @ polyfills.bundle.js:11451invoke @ polyfills.bundle.js:11302
vendor.bundle.js:1229 ERROR Error__defineGetter__: __defineGetter__()__defineSetter__: __defineSetter__()__lookupGetter__: __lookupGetter__()__lookupSetter__: __lookupSetter__()constructor: Error()hasOwnProperty: hasOwnProperty()isPrototypeOf: isPrototypeOf()line: 11837line: 11837message: "Uncaught (in promise): Response with status: 401 Unauthorized for URL: /api/user/me"message: "Uncaught (in promise): Response with status: 401 Unauthorized for URL: /api/user/me"name: "Error"originalStack: "ZoneAwareError@http://127.0.0.1:3000/polyfills.bundle.js:11837:28↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11550:40↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11236:36↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11235:48↵runTask@http://127.0.0.1:3000/polyfills.bundle.js:11003:57↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44"originalStack: "ZoneAwareError@http://127.0.0.1:3000/polyfills.bundle.js:11837:28↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11550:40↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11236:36↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11235:48↵runTask@http://127.0.0.1:3000/polyfills.bundle.js:11003:57↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44"promise: ZoneAwarePromisepromise: ZoneAwarePromisepropertyIsEnumerable: propertyIsEnumerable()rejection: Responserejection: ResponsesourceURL: "http://127.0.0.1:3000/polyfills.bundle.js"sourceURL: "http://127.0.0.1:3000/polyfills.bundle.js"stack: "ZoneAwareError@http://127.0.0.1:3000/polyfills.bundle.js:11837:28↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11550:40↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11236:36↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11235:48↵runTask@http://127.0.0.1:3000/polyfills.bundle.js:11003:57↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44"stack: "ZoneAwareError@http://127.0.0.1:3000/polyfills.bundle.js:11837:28↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11550:40↵resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11236:36↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47↵invokeTask@http://127.0.0.1:3000/polyfills.bundle.js:11235:48↵runTask@http://127.0.0.1:3000/polyfills.bundle.js:11003:57↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44"task: ZoneTasktask: ZoneTasktoLocaleString: toLocaleString()toString: toString()valueOf: valueOf()zone: Zonezone: ZonezoneAwareStack: "resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31 [angular]↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31 [angular]↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47 [angular]↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42 [<root>]↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44 [<root>]"zoneAwareStack: "resolvePromise@http://127.0.0.1:3000/polyfills.bundle.js:11521:31 [angular]↵http://127.0.0.1:3000/polyfills.bundle.js:11598:31 [angular]↵onInvokeTask@http://127.0.0.1:3000/vendor.bundle.js:4133:47 [angular]↵drainMicroTaskQueue@http://127.0.0.1:3000/polyfills.bundle.js:11431:42 [<root>]↵invoke@http://127.0.0.1:3000/polyfills.bundle.js:11302:44 [<root>]"__proto__: Error__proto__: ErrordefaultErrorLogger @ vendor.bundle.js:1229handleError @ vendor.bundle.js:1289next @ vendor.bundle.js:4771(anonymous function) @ vendor.bundle.js:3844__tryOrUnsub @ vendor.bundle.js:26610next @ vendor.bundle.js:26559_next @ vendor.bundle.js:26512next @ vendor.bundle.js:26476next @ vendor.bundle.js:46023emit @ vendor.bundle.js:3830triggerError @ vendor.bundle.js:4202onHandleError @ vendor.bundle.js:4163handleError @ polyfills.bundle.js:11207runGuarded @ polyfills.bundle.js:10979_loop_1 @ polyfills.bundle.js:11442drainMicroTaskQueue @ polyfills.bundle.js:11451invoke @ polyfills.bundle.js:11302
polyfills.bundle.js:10828 TypeError: 'undefined' is not a function (evaluating 'b.style._set(a.propertyName(c),d)')apply @ polyfills.bundle.js:10828(anonymous function) @ polyfills.bundle.js:10828forEach @ (program):1(anonymous function) @ polyfills.bundle.js:10828k @ polyfills.bundle.js:10828(anonymous function) @ polyfills.bundle.js:10828forEach @ (program):1g @ polyfills.bundle.js:10828d @ polyfills.bundle.js:10828
http://localhost:4200/sockjs-node/info?t=1497439031635 Failed to load resource: Could not connect to the server.
vendor.bundle.js:1229 ERROR ErrordefaultErrorLogger @ vendor.bundle.js:1229handleError @ vendor.bundle.js:1289tick @ vendor.bundle.js:5081(anonymous function) @ vendor.bundle.js:4952invoke @ polyfills.bundle.js:11203onInvoke @ vendor.bundle.js:4142invoke @ polyfills.bundle.js:11202run @ polyfills.bundle.js:10963run @ vendor.bundle.js:4010next @ vendor.bundle.js:4952(anonymous function) @ vendor.bundle.js:3844__tryOrUnsub @ vendor.bundle.js:26610next @ vendor.bundle.js:26559_next @ vendor.bundle.js:26512next @ vendor.bundle.js:26476next @ vendor.bundle.js:46023emit @ vendor.bundle.js:3830checkStable @ vendor.bundle.js:4107setHasMicrotask @ vendor.bundle.js:4191onHasTask @ vendor.bundle.js:4154hasTask @ polyfills.bundle.js:11256_updateTaskCount @ polyfills.bundle.js:11276_updateTaskCount @ polyfills.bundle.js:11100runTask @ polyfills.bundle.js:11020drainMicroTaskQueue @ polyfills.bundle.js:11431
polyfills.bundle.js:10828 TypeError: 'undefined' is not a function (evaluating 'b.style._set(a.propertyName(c),d)')apply @ polyfills.bundle.js:10828(anonymous function) @ polyfills.bundle.js:10828forEach @ (program):1(anonymous function) @ polyfills.bundle.js:10828k @ polyfills.bundle.js:10828(anonymous function) @ polyfills.bundle.js:10828forEach @ (program):1g @ polyfills.bundle.js:10828d @ polyfills.bundle.js:10828
vendor.bundle.js:1229 ERROR defaultErrorLogger @ vendor.bundle.js:1229handleError @ vendor.bundle.js:1289tick @ vendor.bundle.js:5081(anonymous function) @ vendor.bundle.js:4952invoke @ polyfills.bundle.js:11203onInvoke @ vendor.bundle.js:4142invoke @ polyfills.bundle.js:11202run @ polyfills.bundle.js:10963run @ vendor.bundle.js:4010next @ vendor.bundle.js:4952(anonymous function) @ vendor.bundle.js:3844__tryOrUnsub @ vendor.bundle.js:26610next @ vendor.bundle.js:26559_next @ vendor.bundle.js:26512next @ vendor.bundle.js:26476next @ vendor.bundle.js:46023emit @ vendor.bundle.js:3830checkStable @ vendor.bundle.js:4107setHasMicrotask @ vendor.bundle.js:4191onHasTask @ vendor.bundle.js:4154hasTask @ polyfills.bundle.js:11256_updateTaskCount @ polyfills.bundle.js:11276_updateTaskCount @ polyfills.bundle.js:11100runTask @ polyfills.bundle.js:11020drainMicroTaskQueue @ polyfills.bundle.js:11431invoke @ polyfills.bundle.js:11302
7polyfills.bundle.js:10828 TypeError: 'undefined' is not a function (evaluating 'b.style._set(a.propertyName(c),d)')apply @ polyfills.bundle.js:10828(anonymous function) @ polyfills.bundle.js:10828forEach @ (program):1(anonymous function) @ polyfills.bundle.js:10828k @ polyfills.bundle.js:10828(anonymous function) @ polyfills.bundle.js:10828forEach @ (program):1g @ polyfills.bundle.js:10828d @ polyfills.bundle.js:10828
```

## What is the motivation / use case for changing the behavior?
Animations should work on ipad in same way

## Please tell us about your environment

Browser:
- [x] Safari (iOS 7) version 11

```
$ ng --version
    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
@angular/cli: 1.0.4
node: 6.10.1
os: linux x64
@angular/animations: 4.2.2
@angular/cli: 1.0.4
@angular/common: 4.2.2
@angular/compiler: 4.2.2
@angular/compiler-cli: 4.2.2
@angular/core: 4.2.2
@angular/flex-layout: 2.0.0-beta.8
@angular/forms: 4.2.2
@angular/http: 4.2.2
@angular/material: 2.0.0-beta.5
@angular/platform-browser: 4.2.2
@angular/platform-browser-dynamic: 4.2.2
@angular/router: 4.2.2
@angular/service-worker: 1.0.0-beta.11
```

### PR DESCRIPTION
**Pull request by matsko**
This patch removes the need to include the Web Animations API Polyfill
(web-animations-js) as a dependency. Angular will now fallback to using
CSS Keyframes in the event that `element.animate` is no longer supported
by the browser.

In the event that an application does use `AnimationBuilder` then the
web-animations-js polyfill is required to enable programmatic,
position-based access to an animation.

Closes #17496

### CODE REVIEW

File: packages/animations/browser/src/render/css_keyframes/css_keyframes_driver.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        element, keyframes, animationName, duration, delay, easing, finalStyles);
+
+    player.onDestroy(() => removeElement(kfElm));
+    return player;
+  }
+
+  private _notifyFaultyScrubber() {
+    if (!this._warningIssued) {
+      console.warn(
+          'ANIMATIONS: please load the web-animations.js polyfill to allow programmatic access');
```
Comment (matsko) [inline packages/animations/browser/src/render/css_keyframes/css_keyframes_driver.ts]: @angular/animations prefix and also include a link to the web-animations-js polyfill (and describe which one).

File: aio/content/guide/animations.md [hunk 2]
```diff
@@ -16,8 +16,10 @@ animation logic with the rest of your application code, for ease of control.
 Angular animations are built on top of the standard [Web Animations API](https://w3c.github.io/web-animations/)
 and run natively on [browsers that support it](http://caniuse.com/#feat=web-animation).
 
-For other browsers, a polyfill is required. Uncomment the `web-animations-js` polyfill from the `polyfills.ts` file.
-
+If the Web Animations API is not supported natively by the browser, then Angular will use CSS keyframes as a
```
Comment (IgorMinar) [inline aio/content/guide/animations.md]: "As of v6, if the Web animations API ..."

File: aio/content/guide/animations.md [hunk 3]
```diff
@@ -16,8 +16,10 @@ animation logic with the rest of your application code, for ease of control.
...
+fallback instead (automatically). This means that the polyfill is no longer required unless any code uses
+[AnimationBuilder](/api/animations/AnimationBuilder). If your code does use AnimationBuilder, then uncomment the
+`web-animations-js` polyfill from the `polyfills.ts` file.
```
Comment (IgorMinar) [inline aio/content/guide/animations.md]: "`polyfills.ts` file generated by Angular CLI."

**Review by IgorMinar — COMMENTED** [review]
can you also update aio/content/guide/browser-support.md? Specifically sections https://next.angular.io/guide/browser-support#enabling-polyfills (don't point to caniuse.org and the src/polyfills.ts comment) and https://next.angular.io/guide/browser-support#optional-browser-features-to-polyfill)

the rest looks good. thanks for all the tests.

**Review by IgorMinar — APPROVED** [review]
I'm going to approve this for now, but mark it for additional cleanup

**Pull request closed without merging**